### PR TITLE
Feature/worker js

### PR DESCRIPTION
**Pull request by my4fdal**
add feature services worker
    1. offline mode site
    2. pedding data request
    3. pending data crud

    replace bugfix firebase message
    1. triger background message
    2. count unread message

    other change file :
            modified:   Commands/BadasoFirebaseCommand.php
            modified:   Commands/BadasoSetup.php
            modified:   Controllers/BadasoDashboardController.php
            modified:   Controllers/BadasoFCMMessagesController.php
            modified:   Helpers/Firebase/FirebasePublishFile.php
            modified:   Routes/api.php
            new file:   resources/js/api/before-request.js
            modified:   resources/js/api/handle-error.js
            modified:   resources/js/api/modules/badaso-fcm.js
            modified:   resources/js/api/resource.js
            modified:   resources/js/app.js
            modified:   resources/js/apps/App.vue
            new file:   resources/js/components/BadasoButtonDataPendingShow.vue
            new file:   resources/js/components/BadasoButtonDataPendingUpdate.vue
            modified:   resources/js/components/BadasoNotificationMessage.vue
            modified:   resources/js/lang/modules/en.js
            modified:   resources/js/lang/modules/id.js
            modified:   resources/js/layout/admin/header/Navbar.vue
            modified:   resources/js/pages/crud-generated/add.vue
            modified:   resources/js/pages/crud-generated/browse.vue
            modified:   resources/js/pages/crud-generated/edit.vue
            modified:   resources/js/pages/crud-management/edit.vue
            new file:   resources/js/pages/data-pending-add/browse.vue
            new file:   resources/js/pages/data-pending-edit/read.vue
            modified:   resources/js/pages/file-manager/browse.vue
            modified:   resources/js/pages/index.vue
            modified:   resources/js/pages/log-viewer/browse.vue
            modified:   resources/js/router/admin/configuration-router.js
            modified:   resources/js/store/modules/badaso.js
            new file:   resources/js/utils/broadcast-messages.js
            new file:   resources/js/utils/check-connection.js
            modified:   resources/js/utils/firebase.js
            new file:   resources/js/utils/indexed-db.js
            modified:   resources/views/admin-panel/index.blade.php

### CODE REVIEW

File: src/resources/js/components/BadasoButtonDataPendingUpdate.vue [hunk 1]
```diff
@@ -0,0 +1,47 @@
+<template>
```
Comment (rizkiheryandi) [inline src/resources/js/components/BadasoButtonDataPendingUpdate.vue:1]: is create a component for this not an overkill ?

maybe it's better create directly on the related page

File: src/Controllers/BadasoDashboardController.php [hunk 2]
```diff
@@ -14,6 +14,7 @@ class BadasoDashboardController extends Controller
 {
     public function verifyLicense()
     {
+        return ApiResponse::success([]);
```
[the remark this answers is not in the document]
Comment (my4fdal) [reply src/Controllers/BadasoDashboardController.php:17]: sorry, i will fix it

File: src/Commands/BadasoFirebaseCommand.php [hunk 3]
```diff
@@ -12,7 +12,7 @@ class BadasoFirebaseCommand extends Command
      *
      * @var string
      */
-    protected $name = 'badaso:firebase publish';
```
[the remark this answers is not in the document]
Comment (my4fdal) [reply src/Commands/BadasoFirebaseCommand.php:15]: especially if there is a change to the firebase key api, for badaso: setup is added too

File: src/resources/js/components/BadasoButtonDataPendingUpdate.vue (hunk 1 above)
Comment (my4fdal) [reply src/resources/js/components/BadasoButtonDataPendingUpdate.vue:1]: ok, i will fix it

**Review by rizkiheryandi — APPROVED** [review]
lgtm

**Pull request merged**